RecoHGCal/TICL/interface/ClusterFilterByAlgo.h

```diff
@@ -0,0 +1,35 @@
+// Author: Marco Rovere - marco.rovere@cern.ch
+// Date: 11/2018
+// Copyright CERN
+
+#ifndef RecoHGCal_TICL_ClusterFilterByAlgo_H__
+#define RecoHGCal_TICL_ClusterFilterByAlgo_H__
+
+#include "DataFormats/CaloRecHit/interface/CaloCluster.h"
+#include "RecoHGCal/TICL/interface/ClusterFilterBase.h"
+
+#include <memory>
+#include <utility>
+
+// Filter clusters that belong to a specific algorithm
+class ClusterFilterByAlgo final : public ClusterFilterBase {
+ public:
+  ClusterFilterByAlgo(const edm::ParameterSet& ps)
+      : ClusterFilterBase(ps), algo_number_(ps.getParameter<int>("algo_number")) {}
```

Comment:
will this "algo_number" be centrally enforced somehow? It seems like it could cause confusion if not.